messaging/Controllers/BundlesController.cs

```diff
@@ -209,14 +207,34 @@ public async Task<ActionResult<Bundle>> PostIncomingMessageItem(string jurisdict
                 _logger.LogError("Rejecting request with invalid jurisdiction ID.");
                 return BadRequest("Invalid jurisdiction ID");
             }
+
+            Bundle bundle = BaseMessage.ParseGenericBundle(text.ToString(), true);
+
+            if (bundle.Type.Equals(Bundle.BundleType.Message))
+            {
+                if (!ValidateJurisdictionId(bundle, jurisdictionId))
+                {
+                    return BadRequest("The message resource jurisdiction ID did not match the parameter jurisdiction ID [" + jurisdictionId + "].");
+                }
+            }
+            else if (bundle.Type.Equals(Bundle.BundleType.Batch))
+            {
+                foreach (Bundle.EntryComponent entry in bundle.Entry)

```

Comment:
When the submission is a bundle, we send back a batch response with response codes for each message. If one message fails the jurisdiction validation, only that message should return a 400. The other messages could be successful and return 200. Instead of having this validation check at the beginning, these checks should be moved into the existing validation so we can maintain the same pattern

Reply:
The validation should occur in ParseIncomingMessageItem and throw an ArgumentException. That should return the correct response code for a single message submission and a batch